frontend/src/components/commentSection/index.js

```diff
@@ -80,7 +96,33 @@ class CommentSection extends Component {
       return comment.parent_comment_id === null
     })
 
-    return filteredComments
+    const userComments = []
+    const pinnedComments = []
+    const unpinnedComments = []
+
+    filteredComments.forEach((comment) => {
+      if (comment.profile.display_name === this.props.store.profile.display_name) {
+        userComments.push(comment)
+      } else if (comment.is_pinned) {
+        pinnedComments.push(comment)
+      } else {
+        unpinnedComments.push(comment)
+      }
+    })
+
+    const fullyFilteredSortedComments =
+      userComments.sort ((firstComparedUserComment, secondComparedUserComment) => {
+        return new Date(firstComparedUserComment.created_at).getTime() <
+          new Date(secondComparedUserComment.created_at).getTime() ? 1 : -1
+      }).concat(pinnedComments.sort((firstComparedPinnedComment, secondComparedPinnedComment) => {
```

Comment:
A piece like this could be

```suggestion
      userComments.sort((currComment, nextComment) => (
        new Date(currComment.created_at) - new Date(nextComment.created_at)
      ))
```

No need to do getTime and a ternary and also a curly brace with a return instead of a direct return arrow func.

Reply:
You can take this and put it in a function called something like `sortByDatetime` which you can used for all comments. Would look like this:

```js
sortArrayByDatetime = (arrToSort, dateProperty='created_at') => (
  arrToSort.sort((currItem, nextItem) => (
    new Date(currItem[dateProperty]) - new Date(nextItem[dateProperty])
  ))
)
```

Just a concept. Then you could just do something like
```js
[
  ...sortArrayByDatetime(userComments),
  ...sortArrayByDatetime(pinnedComments),
  ...sortArrayByDatetime(unpinnedComments)
]
```

A lot cleaner, don't you think?

Reply:
You might even get away with not doing `new Date()` at all for the sorting

Just `currItem[dateProperty] - nextItem[dateProperty]`. Pretty sure that works too but test it